## 47장

### 에러의 전파?
  
    에러는 호출자 방향으로 전파된다. 즉, 콜 스택의 아래방향으로 전파된다.
    
```js
const foo = () => {
  throw Error('foo에서 발생한 에러'); // 4
}

const bar = () => {
  foo(); // 3
}

const baz = () => {
  bar(); //2
}

try {
  baz() // 1
} catch (err) {
  console.error(err);
}
```
    실행 컨텍스트 스택 : 전역 실행 컨텍스트 > baz 실행 컨텍스트 > bar 실행 컨텍스트 > foo 실행 컨텍스트
    
<br>
    
    foo에서 에러가 발생했는데 
    foo 실행 컨텍스트에서는 에러를 캐치하지 않으니 bar로 에러 전파.
    bar -> baz -> 전역의 catch문에서 에서 캐치한다.
    
    throw된 에러를 캐치하여 적절히 대응하면 
    프로그램을 강제 종료시키지 않고 코드의 실행 흐름을 복구할 수 있다. 
    throw된 에러를 어디에서도 캐치하지 않으면 프로그램은 강제 종료된다.

<br>

## 48장
